packages/client/src/utils/buttonActions.js

```diff
@@ -514,6 +514,10 @@ const rowActionHandler = async action => {
   })
 }
 
+const delayHandler = async action => {
+  return new Promise(resolve => setTimeout(resolve, action.parameters.duration))
```

Comment:
This will work regardless of what value `duration` is, but we should probably sanity check it anyway. There's also no guarantee that it's a number - so I would use a normal number input for the setting instead of a drawer bindable input, and also do `action.parameters.duration || 0` here to account for null/undefined/empty string.